### PR TITLE
more ?? and ??= documentation

### LINKED ISSUE #712: [Undocumented] Operator `??=`
#711 documents operator `??=`

### LINKED ISSUE #728: Add more examples to coalesce operator documentation
Please complete the information below:

# Where is the problem?

Details of using coalesce operator. Not enough examples.

# What is the problem?

It does not explain usages with index lookups. I've had to explain multiple times (to other FB co-workers) how this works and usage patterns. Specifically how it compares to `idx()`.  It would save a lot of time If I there was accurate documentation that removed all doubt about how it works. Thank you.

--------------------------------

Please don't change anything below this point.

--------------------------------

 - Build ID: HHVM=HHVM-4.23.0:HSL=v4.15.2:2019-09-20T21:41:50+0000:138dd8d9d3c423f9864bf5952632efca10d6696c
 - Page requested: /hack/expressions-and-operators/coalesce
 - Page requested at: Tue, 24 Sep 2019 20:13:13 +0000
 - Controller: GuidePageController

### PR DESCRIPTION
**Pull request by jjergus**
+ comparison with idx()
+ new section and example for ??=

Fixes #728
Fixes #712

### CODE REVIEW

**Comment by fredemmott** [comment]
How should this stack with https://github.com/hhvm/user-documentation/pull/711 ?

**Comment by jjergus** [comment]
Good catch. I integrated everything from #711 into this one.

File: guides/hack/03-expressions-and-operators/91-coalesce-examples/assignment.hack [hunk 1]
```diff
@@ -0,0 +1,38 @@
+use namespace HH\Lib\{Dict, Str};
```
Comment (fredemmott) [inline guides/hack/03-expressions-and-operators/91-coalesce-examples/assignment.hack]: It'd be nice to declare a namespace here; most examples do, and it avoids polutting the root namespace which can get annoying in the actual site code

File: guides/hack/03-expressions-and-operators/91-coalesce.md [hunk 2]
```diff
@@ -1,9 +1,68 @@
...
+Given the expression `e1 ?? e2`, if `e1` is defined and not `null`, then the
+result is `e1`. Otherwise, `e2` is evaluated, and its value becomes the result.
+There is a sequence point after the evaluation of `e1`.
 
 ```Hack
-$arr = dict["black" => 10, "white" => null];
-$arr["black"] ?? -100  // 10 as $arr["black"] is defined and not null
-$arr["white"] ?? -200  // -200 as $arr["white"] is null
-$arr["blue"] ?? -300   // -300 as $arr["blue"] is not defined
+$nully = null;
```
Comment (fredemmott) [inline guides/hack/03-expressions-and-operators/91-coalesce.md]: would also be nice to split this to a file (And the other inline ones)

**Pull request merged**